app/src/main/java/com/example/facebookapp/activity/SignUpActivity.java

```diff
@@ -0,0 +1,183 @@
+import android.content.SharedPreferences;
+import android.os.Bundle;
+import android.text.Editable;
+import android.text.TextWatcher;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.ProgressBar;
+import android.widget.Toast;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.example.facebookapp.R;
+import com.example.facebookapp.model.AccountModel;
+import com.example.facebookapp.network.ApiService;
+import com.example.facebookapp.network.ResponseCode;
+import com.example.facebookapp.network.RetrofitClient;
+import com.google.android.material.textfield.TextInputLayout;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+public class SignUpActivity extends AppCompatActivity {
+
+    private static final String TAG = "SignUpActivity.this";
+    private static final String NOT_ENOUGH = "Not enough information";
+    private static final String FORMAT_ERROR = "Format error";
+
+    private ApiService mService;
+    ProgressBar progressBar;
+
+    private String phone = "";
+    private String password = "";
+    private String confirmPassword = "";
+
+    private EditText editPhone, editPassword, editConfirmPassword;
+    private TextInputLayout editPhoneLayout, editPasswordLayout, editConfirmLayout;
+    private Button buttonSignUp;
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_signup);
+
+        editPhoneLayout = findViewById(R.id.edit_phone_signUp_textLayout);
+        editPasswordLayout = findViewById(R.id.edit_password_signUp_textLayout);
+        editConfirmLayout = findViewById(R.id.edit_confirm_password_textLayout);
+        editPhone = findViewById(R.id.edit_phone_signUp);
+        editPassword = findViewById(R.id.edit_password_signUp);
+        editConfirmPassword = findViewById(R.id.edit_confirm_password);
+        buttonSignUp = findViewById(R.id.button_signUp);
+        mService = RetrofitClient.getInstance().create(ApiService.class);
+    }
+
+    @SuppressLint("CommitPrefEdits")
+    @Override
+    protected void onResume() {
+        super.onResume();
+        SharedPreferences sharedPreferences = getSharedPreferences("SignUpPrefs", MODE_PRIVATE);
+        SharedPreferences.Editor editor = sharedPreferences.edit();
+
+        editPhone.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+            }
+
+            @Override
+            public void onTextChanged(CharSequence s, int start, int before, int count) {
+                String tempPhone = s.toString();
+                if (tempPhone.charAt(0) != '0' || tempPhone.length() > 10)
+                    editPhoneLayout.setError(FORMAT_ERROR);
+                else
+                    editPhoneLayout.setError(null);
```

Comment:
khi em xóa hết thì charAt(0) nó sẽ bị outIndexBound vì k có kí tự thứ 0 
Em nên dùng Regex để validate các trường nhập 
ở đây để validate phone em tham khảo ở đây https://www.regextester.com/98578
https://stackoverflow.com/questions/22505336/email-and-phone-number-validation-in-android